Record a warning type event

FengyunPan opened the pull request:
A warning type event should be recorded when failed to calculate
the number of expected pods.

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/44374)
<!-- Reviewable:end -->

NickrenREN commented:
/assign

NickrenREN commented:
/lgtm

NickrenREN commented:
/release-note-none

NickrenREN commented:
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[pkg/controller/disruption/disruption.go, line 495 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/44374#-KhV4cVIve12t-EXiv-i:-KhV4cVIve12t-EXiv-j:b-gt44k3) ([raw file](https://github.com/kubernetes/kubernetes/blob/cbb673d7d92b410ab523bbb1b486fecbf1013fbc/pkg/controller/disruption/disruption.go#L495)):*
> ```Go
> 	expectedCount, desiredHealthy, err := dc.getExpectedPodCount(pdb, pods)
> 	if err != nil {
> 		dc.recorder.Eventf(pdb, v1.EventTypeWarning, "ExpectedPods", "Failed to calculate the number of expected pods: %v", err)
> ```

change reason("ExpectedPods") to show the err ?

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/44374)*
<!-- Sent from Reviewable.io -->

Commit:
Record a warning type event

A warning type event should be recorded when failed to calculate
the number of expected pods.
And the same to daemoncontroller when failed to place pod.

FengyunPan commented:
@NickrenREN Thank for your review, I will change reason soon.

deads2k commented:
/approve

FengyunPan commented:
no bug @k8s-bot non-cri e2e test this
